pkg/webhook/core/mutating_handler.go

```diff
@@ -0,0 +1,83 @@
+package core
+
+import (
+	"context"
+	"encoding/json"
+	admissionv1 "k8s.io/api/admission/v1"
+	"net/http"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var mutatingHandlerLog = ctrl.Log.WithName("mutating_handler")
+
+type mutatingHandler struct {
+	mutator Mutator
+	decoder *admission.Decoder
+}
+
+func (h *mutatingHandler) SetDecoder(d *admission.Decoder) {
+	h.decoder = d
+}
```

Comment:
The biggest difference is that the other repos are not working off the v1 spec for admission control and include an API to "inject" a decoder. Instead, we have to create the decoder manually.